File: design-pattern-go/src/patterns/structural/flyweight-cache/ColorShape.go
package flyweightcache

type ColorShape struct {
	color string
	shape string
}

func NewColorShape(color string, shape string) ColorShape {
	return ColorShape{color: color, shape: shape}
}

func (cs *ColorShape) Move(context *Particle, x int, y int) *Particle {
	context.x = x
	context.y = y
	return context
}

File: design-pattern-java/app/src/main/java/design/pattern/java/patterns/structural/flyweight_cache/ColorShape.java
package design.pattern.java.patterns.structural.flyweight_cache;

public class ColorShape {
  protected final String color;
  protected final String shape;

  public ColorShape(String color, String shape) {
    this.color = color;
    this.shape = shape;
  }

  public Particle move(Particle context, Number x, Number y) {
    return context.setX(x).setY(y);
  }
}
